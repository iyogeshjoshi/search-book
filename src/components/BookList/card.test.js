import React from "react";
import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Card from "./card";

describe("Renders Card correctly", () => {
  afterAll(cleanup);
  it("renders book card", () => {
    const properties = {
      index: 1,
      book: {
        title: "sample title",
        author: {
          id: 2,
          author: "Some author",
        },
        summary: "This is some summary",
      },
    };
    // const { getByText } = render(<Card {...properties} />);
    const card = renderer.create(<Card {...properties} />).toJSON();

    expect(card).toMatchSnapshot();
  });
  it("renders with attributes", () => {
    const properties = {
      index: 1,
      book: {
        title: "sample title",
        author: {
          id: 2,
          author: "Some author",
        },
        summary: "This is some summary",
      },
    };
    const { getByText } = render(<Card {...properties} />);

    expect(getByText(/sample title/i)).toBeTruthy();
    expect(getByText(/some author/i)).toBeTruthy();
    expect(getByText(/This is some summary/i)).toBeTruthy();
  });
});
