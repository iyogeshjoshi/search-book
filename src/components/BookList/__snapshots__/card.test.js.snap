// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Card correctly renders book card 1`] = `
<div
  className="book-card"
>
  <div
    className="title"
    title="sample title"
  >
    sample title
  </div>
  <div
    className="author"
  >
    By 
    Some author
  </div>
  <div
    className="summary"
    title="This is some summary"
  >
    This is some summary
  </div>
</div>
`;
